client/src/components/Ticketing/ticketingmanager/DoorList/DoorList.tsx

```diff
@@ -6,72 +6,71 @@ import {titleCase} from '../../../../utils/arrays';
-      audience: process.env.REACT_APP_ROOT_URL,
-      scope: 'admin',
-    });
-    const res = await fetch(
-      process.env.REACT_APP_API_2_URL + `/events/checkin`,
-      {
-        credentials: 'include',
-        method: 'PUT',
-        headers: {
-          'Content-Type': 'application/json',
-          'Authorization': `Bearer ${token}`,
-        },
-        body: JSON.stringify({isCheckedIn, ticketID}),
-      },
-    );
-    if (!res.ok) {
-      throw new Error(`Failed to check in guest. HTTP status: ${res.status}`);
-    }
-    return res.json();
-  } catch (err) {
-    console.error(err.message);
-  }
-};
 
 const renderCheckbox = (params: GridCellParams) => (
   <Checkbox checked={params.value as boolean} disabled color='info' />
 );
 
-/**
- * renders in the check in for guests
- *
- * @param {GridCellParams} params
- * @returns edits the checkInGuest value
- */
-const renderCheckin = (params: GridCellParams) => {
-  const isDisabled = params.getValue(params.id, 'lastname') === 'OPEN SEATS';
+interface RenderCheckinProps {
+  params: GridCellParams;
+}
+
+const RenderCheckin = (props: RenderCheckinProps) => {
+  const {params} = props;
+  const {token} = useFetchToken();
+  const [checked, setChecked] = useState(params.value as boolean);
+  const checkInGuest = async (isCheckedIn: boolean, rowId: string) => {
+      const [contactId, instanceId] = rowId.split('-');
+      const res = await fetch(
+          process.env.REACT_APP_API_2_URL + `/events/checkin`,
+          {
+            credentials: 'include',
+            method: 'PUT',
+            headers: {
+              'Content-Type': 'application/json',
+              'Authorization': `Bearer ${token}`,
+            },
+            body: JSON.stringify({isCheckedIn, contactId, instanceId}),
+          },
+      );
+      if (!res.ok) {
+        throw new Error(`Failed to update guest checkin status`);
+      }
+  };
 
+  if (token === '') return null;
   return (
     <Checkbox
       color='primary'
-      defaultChecked={params.value as boolean}
-      disabled={isDisabled}
-      onChange={(e) => {
-        if (!isDisabled) {
-          checkInGuest(
-            e.target.checked,
-            params.getValue(params.id, 'ticketno') as string,
-          );
-        }
+      checked={checked}
+      onChange={() => {
+        checkInGuest(
+            !checked,
+            params.row.id,
+        )
+            .then(() =>
+                setChecked((prev) => !prev))
+            .catch((error) => console.error(error));
       }}
     />
   );
 };
 
+const renderTicketTypes = (params: GridCellParams) => {
+  const ticketTypes = params.value;
+  return (
+      <ul className={'w-full h-[80%] overflow-scroll'}>
+        {
+          typeof ticketTypes === 'object' &&
+            Object.keys(ticketTypes).map((key, index) => (
+                <li key={`${params.row.id}-${index}`}>{key} - {ticketTypes[key]}</li>
+            ))
+        }
+      </ul>
+  );
+};
```

Comment:
This looks great, nice solution! I also did not realize how robust that search bar already is in that you can toggle which columns are displayed and much more, definitely we'll want the same kind of thing in the contacts search. A couple of things though - the blocking one is that when exporting to excel, the ticket type(s) show up as `[object Object]`, and the less important one is that it would be great if it was possible to resize each column. for instance, my email ben16williams@gmail.com is too long and gets cut off so that the only way I can see the whole value is by hovering over it with my mouse (I could be being dumb and missing how you're suppose to do this though)

Reply:
Ooof of course it is. The name, email and ticket types are probably the most important text fields that should have enough space

Reply:
Fixed